### ISSUE TITLE
bug(SingleSelect & MultiSelect): select options hidden behind modals

### ISSUE THREAD

**Issue by calebchiam** [opengovsg/design-system#109]
**Describe the bug**
When using the `MultiSelect` component in a `Modal`, the options are hidden behind the modal.

This bug seems to have been introduced with #106, since it was working fine before the change.

**To Reproduce**
In a standard Chakra `Modal` component,

```
<Modal isOpen={isOpen} onClose={onClose}>
  <ModalContent>
    <ModalHeader>I am a modal</ModalHeader>
    <ModalBody>
      <MultiSelect
        name="selection"
        placeholder={'Please select'}
        values={[]}
        items={[...Array(20).keys()].map((idx) => {
          return { label: `select me ${idx}`, value: `select me ${idx}` }
        })}
      />
    </ModalBody>
    <ModalFooter>Some stuff below</ModalFooter>
  </ModalContent>
</Modal>
```

Click on the input box to see the dropdown options, they will render behind the modal.

**Expected behavior**
Select options should render in front of the modal.

**Screenshots**
![image](https://user-images.githubusercontent.com/14286996/198922798-0c5ae7a8-4d8d-4455-9bb2-0b00ebbf96b8.png)

**Comment by njenwei** [comment]
Could be this https://github.com/opengovsg/design-system/blob/a52c8995c552a7764bc699939776af19edbb7af3/react/src/Dropdown/components/MultiSelectMenu.tsx#L33

**Comment by calebchiam** [comment]
Tested with a local build, doesn't seem to be a `zIndex` issue. 

The same issue of options hiding behind modals occurs for regular `SingleSelect` components as well, so I've updated the issue title. (And the `SingleSelect` component doesn't specify `zIndex`.)

For both `SingleSelect` and `MultiSelect`, deleting the `FloatingPortal` wrapper seemed to resolve the issue. `FloatingPortal` was introduced as part of https://github.com/opengovsg/FormSG/pull/5137 -- not sure what the implications are if it's deleted.

**Comment by karrui** [comment]
Fixed, just not pushed. Added zIndex prop to ChakraProvider. Should handle portal nesting just fine now (hopefully).

if you don't want to wait, and don't want to handle copy pasted code (and use back current design system code), you just need to add `portalZIndex={x)` (where x is applicable. 40 seems fine haha) to your current `ChakraProvider`.

See https://github.com/chakra-ui/chakra-ui/issues/3269#issuecomment-774484028

**Comment by karrui** [comment]
fixed in latest `v0.1.0-alpha.0` release
